File: app/src/main/java/ir/jashakouri/opencvproject/view/base/BaseActivity.java
package ir.jashakouri.opencvproject.view.base;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.WindowManager;
import android.widget.Toast;

import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.kongqw.permissionslibrary.PermissionsManager;

import ir.jashakouri.opencvproject.utils.StatusBarUtils;

public abstract class BaseActivity extends AppCompatActivity {

    protected PermissionsManager mPermissionsManager;

    @LayoutRes
    protected abstract int layout();

    protected abstract void init(Bundle savedInstanceState);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
        setContentView(layout());
        init(savedInstanceState);
        StatusBarUtils.setLightStatusBar(this);
    }

    /**
     * Review authority
     *
     * @param requestCode  requestCode
     * @param permissions  permissions
     * @param grantResults grantResults
     */
    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        mPermissionsManager.recheckPermissions(requestCode, permissions, grantResults);
    }

    protected void showPermissionDialog() {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setTitle("Request permission");
        builder.setMessage("Application need request camera, storage, voice recorder permissions");
        builder.setPositiveButton("Go to set permissions", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
                PermissionsManager.startAppSettings(getApplicationContext());
            }
        });
        builder.setNegativeButton("cancel", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
            }
        });
        builder.create().show();
    }

    protected void showInstallDialog() {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setCancelable(false);
        builder.setTitle("You have not installed OpenCV Manager");
        builder.setMessage("Whether to download and install?");

        builder.setPositiveButton("Download", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                Toast.makeText(getApplicationContext(), "Downloading", Toast.LENGTH_SHORT).show();
                dialog.dismiss();
                startActivity(new Intent(Intent.ACTION_VIEW).setData(Uri.parse("https://github.com/kongqw/FaceDetectLibrary/tree/opencv3.2.0/OpenCVManager")));
            }
        });
        builder.setNegativeButton("cancel", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
                finish();
            }
        });
        builder.create().show();
    }

}

File: app/src/main/java/ir/jashakouri/opencvproject/view/base/BaseActivity.kt
package ir.jashakouri.opencvproject.view.base

import android.content.Intent
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.view.View
import android.view.WindowManager
import android.widget.Toast
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.kongqw.permissionslibrary.PermissionsManager

abstract class BaseActivity : AppCompatActivity() {

    protected var mPermissionsManager: PermissionsManager? = null

    @LayoutRes
    abstract fun layout(): Int

    abstract fun init(savedInstanceState: Bundle?)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layout())
        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
        init(savedInstanceState)
    }

    protected fun hideStatusBar(){

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            window.decorView.systemUiVisibility = (View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                    or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                    or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                    or View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                    or View.SYSTEM_UI_FLAG_FULLSCREEN
                    or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY)
        }

    }

    /**
     * Review authority
     *
     * @param requestCode  requestCode
     * @param permissions  permissions
     * @param grantResults grantResults
     */
    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<String?>,
        grantResults: IntArray
    ) {
        // After the user makes a choice, review the permissions to determine whether the permission application is passed
        mPermissionsManager!!.recheckPermissions(requestCode, permissions, grantResults)
    }

    /**
     * Show missing permission dialog
     */
    protected open fun showPermissionDialog() {
        val builder = AlertDialog.Builder(this)
        builder.setTitle("Request permission")
        builder.setMessage("Android 6.0+ dynamically request camera permissions")
        builder.setPositiveButton(
            "Go to set permissions"
        ) { dialog, _ ->
            dialog.dismiss()
            PermissionsManager.startAppSettings(applicationContext)
        }
        builder.setNegativeButton(
            "cancel"
        ) { dialog, _ -> dialog.dismiss() }
        builder.create().show()
    }

    /**
     * Show a dialog box without OpenCV Manager installed
     */
    protected open fun showInstallDialog() {
        val builder = AlertDialog.Builder(this)
        builder.setCancelable(false)
        builder.setTitle("You have not installed OpenCV Manager")
        builder.setMessage("Whether to download and install?")
        builder.setPositiveButton(
            "Download"
        ) { dialog, which ->
            Toast.makeText(applicationContext, "Downloading", Toast.LENGTH_SHORT).show()
            dialog.dismiss()
            startActivity(Intent(Intent.ACTION_VIEW).setData(Uri.parse("https://github.com/kongqw/FaceDetectLibrary/tree/opencv3.2.0/OpenCVManager")))
        }
        builder.setNegativeButton(
            "Exit"
        ) { dialog, which ->
            dialog.dismiss()
            finish()
        }
        builder.create().show()
    }

}